google/cloud/opentelemetry/internal/monitored_resource.cc

```diff
@@ -229,6 +233,52 @@ MonitoredResource ToMonitoredResource(
   return provider.Process(attributes);
 }
 
+opentelemetry::sdk::metrics::ResourceMetrics
+GetCopyWithServiceResourceLabelsAsMetricLabels(
+    opentelemetry::sdk::metrics::ResourceMetrics const& data) {
+  if (!data.resource_) {
+    return data;
+  }
+
+  std::unordered_set<std::string> const& kOtelServiceKeys{
+      sc::kServiceName, sc::kServiceNamespace, sc::kServiceInstanceId};
+
+  std::unordered_map<std::string,
+                     opentelemetry::sdk::common::OwnedAttributeValue>
+      service_labels;
+  auto resource_attributes = data.resource_->GetAttributes();
+  for (auto it = resource_attributes.begin(); it != resource_attributes.end();
+       ++it) {
+    if (kOtelServiceKeys.find(it->first) != kOtelServiceKeys.end()) {
+      service_labels[it->first] = it->second;
+    }
+  }
+
+  if (service_labels.empty()) {
+    return data;
+  }
+
+  opentelemetry::sdk::metrics::ResourceMetrics data_copy = data;
+  for (opentelemetry::sdk::metrics::ScopeMetrics& scope_metrics :
+       data_copy.scope_metric_data_) {
+    for (opentelemetry::sdk::metrics::MetricData& metric :
+         scope_metrics.metric_data_) {
+      for (opentelemetry::sdk::metrics::PointDataAttributes& pda :
+           metric.point_data_attr_) {
+        auto& attributes = pda.attributes;
```

Comment:
Sorry, I am confused. Do we want the labels to end up getting added to the `google.api.Metric` https://github.com/googleapis/googleapis/blob/2d05911be5a468b556236bee537c91922f9c23a3/google/api/metric.proto#L286

... or to the `google.api.MonitoredResource` https://github.com/googleapis/googleapis/blob/2d05911be5a468b556236bee537c91922f9c23a3/google/api/monitored_resource.proto#L106

... or both?

unit tests would help clarify the intended behavior that we want.